Initial port

PaulTerrasi opened the pull request:
@michaelsilver The goal of this pull request is to get all of our Swerve Drive code into this repo and structure it in a way that makes sense for a standalone project. This should be pretty obvious from the code, but this project isn't intended to be deployed to a robot because it lacks an OI. Instead, it is just a collection of classes to be copied into any future robot projects.

I know this is a lot of code, but right now I don't need you to scrutinize it. I will be doing a lot of clean up and commenting in future pull requests. I just want you to look through and make sure that the structure makes sense, and that you don't see any bugs that have cropped up in the transfer.

Closes #1

michaelsilver commented:
Thank you for doing this. Overall, `org.usfirst.frc.team246.robot.commands`, `org.usfirst.frc.team246.robot.OverclockedLibraries`, and `org.usfirst.frc.team246.nav6` make sense to me, but the conglomerate in `org.usfirst.frc.team246.robot` seems really seems random to me -- `SwerveModule.java`, `Vector2D.java`, etc.. Why not have these back in overclockedLibraries like they were? And where's `RobotMap.java`?

I'm sure you had reasons for this, and if you are convinced this is the right way to set up the libraries, I'd like you to please create a new section in the README (best way to edit it is directly on GitHub website, but make sure you are in the right branch) where you explain to a novice FRC programmer how to use this repository to setup their OverclockedSwerve based FRC project.

PaulTerrasi commented:
I can see what your saying about the randomness of files in `org.usfirst.frc.team246.robot`. The idea was these are the classes someone integrating this into their own project will likely have to interact with directly. But, on second thought, I feel like it is more confusing than it is helpful. I have moved everything but Robot, Drivetrain, and SwerveModule into OverclockedLibraries.

There is no RobotMap because this is not meant to actually deploy to a robot. I added some constants into Robot.java to use in the commands (which, besides SteeringWatchdog, DrivingCommand, and FieldCentricDrivingCommand, are mainly intended to be examples), but I don't want to go any further by creating an actually creating a RobotMap and OI.

The explanation of how to add this to your project will be done inside of the whitepaper, which I will link to inside of README when I am done.

Also, I am not sure if you did this, but I do actually want you to read through the code, because I did change a lot to make this thing stand alone. I just don't need you to comment on the stuff that I have already made issues about, which includes making the code look pretty. To help narrow things down, you really only need to review the files in `org.usfirst.frc.team246.robot`, `org.usfirst.frc.team246.robot.commands`, and `PIDConstants.java`.

michaelsilver commented:
LGTM! :rocket:
